test/uk/gov/hmrc/helptosavefrontend/controllers/RegisterControllerSpec.scala

```diff
@@ -313,18 +314,106 @@ class RegisterControllerSpec
+
+        "show the details page for valid users" in {
+          inSequence {
+            mockAuthWithNINORetrievalWithSuccess(AuthWithCL200)(mockedNINORetrieval)
+            mockEnrolmentCheck()(Right(EnrolmentStatus.NotEnrolled))
+            mockSessionCacheConnectorGet(Right(Some(HTSSession(Some(Right(randomEligibleWithUserInfo(validUserInfo))), Some("valid@email.com"), None, None, None, Some(bankDetails)))))
+          }
           val result = doRequest()
 
           status(result) shouldBe 200
-          contentAsString(result) should include("Check your details")
+          contentAsString(result) should include("check your details")
         }
 
-      }
+        "redirect existing account holders to NS&I" in {
+          inSequence {
+            mockAuthWithNINORetrievalWithSuccess(AuthWithCL200)(mockedNINORetrieval)
+            mockEnrolmentCheck()(Right(EnrolmentStatus.Enrolled(true)))
+          }
+          val result = doRequest()
+
+          status(result) shouldBe 303
+          redirectLocation(result) shouldBe Some("https://nsandi.com")
+        }
+
+        "redirect user to eligibility check if there is no active session found" in {
+          inSequence {
+            mockAuthWithNINORetrievalWithSuccess(AuthWithCL200)(mockedNINORetrieval)
+            mockEnrolmentCheck()(Right(EnrolmentStatus.NotEnrolled))
+            mockSessionCacheConnectorGet(Right(None))
+          }
+          val result = doRequest()
+
+          status(result) shouldBe 303
+          redirectLocation(result) shouldBe Some(routes.EligibilityCheckController.getCheckEligibility().url)
+        }
+
+        "redirect user to eligibility check if session exists but no eligibility result" in {
+          inSequence {
+            mockAuthWithNINORetrievalWithSuccess(AuthWithCL200)(mockedNINORetrieval)
+            mockEnrolmentCheck()(Right(EnrolmentStatus.NotEnrolled))
+            mockSessionCacheConnectorGet(Right(Some(HTSSession(None, Some("valid@email.com"), None, None, None, Some(bankDetails)))))
+          }
+          val result = doRequest()
+
+          status(result) shouldBe 303
+          redirectLocation(result) shouldBe Some(routes.EligibilityCheckController.getCheckEligibility().url)
+        }
+
+        "show user not eligible page if the user is not eligible" in {
+          inSequence {
+            mockAuthWithNINORetrievalWithSuccess(AuthWithCL200)(mockedNINORetrieval)
+            mockEnrolmentCheck()(Right(EnrolmentStatus.NotEnrolled))
+            mockSessionCacheConnectorGet(Right(Some(HTSSession(Some(Left(randomIneligibility())), None, None, None, None, Some(bankDetails)))))
+          }
+          val result = doRequest()
+
+          status(result) shouldBe 303
+          redirectLocation(result) shouldBe Some(routes.EligibilityCheckController.getIsNotEligible().url)
+        }
+
+        "handle the case when user doesn't have a valid GG email in the session" in {
```

Comment:
there may be a case where someone doesn't have a GG email but has reached this point in the journey by entering an email via the enter-email page